homeassistant/components/climate/__init__.py

```diff
@@ -345,56 +349,58 @@ def _report_turn_on_off(feature: str, method: str) -> None:
-                self.__mod_supported_features |= (  # pylint: disable=unused-private-member
-                    ClimateEntityFeature.TURN_OFF
-                )
-            elif self.hvac_modes and HVACMode.OFF in self.hvac_modes:
-                # turn_off implicitly supported by including HVACMode.OFF
-                _report_turn_on_off("off", "turn_off")
-                self.__mod_supported_features |= (  # pylint: disable=unused-private-member
-                    ClimateEntityFeature.TURN_OFF
-                )
+        if self._enable_turn_on_off_backwards_compatibility is False:
+            # Return if integrations have turned off backwards compatibility
+            return
 
-        if not self.supported_features & ClimateEntityFeature.TURN_ON:
-            if (
-                type(self).async_turn_on is not ClimateEntity.async_turn_on
-                or type(self).turn_on is not ClimateEntity.turn_on
-            ):
-                # turn_on implicitly supported by implementing turn_on method
-                _report_turn_on_off("TURN_ON", "turn_on")
-                self.__mod_supported_features |= (  # pylint: disable=unused-private-member
-                    ClimateEntityFeature.TURN_ON
-                )
-            elif self.hvac_modes and any(
-                _mode != HVACMode.OFF and _mode is not None for _mode in self.hvac_modes
-            ):
-                # turn_on implicitly supported by including any other HVACMode than HVACMode.OFF
-                _modes = [_mode for _mode in self.hvac_modes if _mode != HVACMode.OFF]
-                _report_turn_on_off(", ".join(_modes or []), "turn_on")
-                self.__mod_supported_features |= (  # pylint: disable=unused-private-member
-                    ClimateEntityFeature.TURN_ON
-                )
+        if not self.supported_features & ClimateEntityFeature.TURN_OFF and (
+            type(self).async_turn_off is not ClimateEntity.async_turn_off
+            or type(self).turn_off is not ClimateEntity.turn_off
+        ):
+            # turn_off implicitly supported by implementing turn_off method
+            _report_turn_on_off("TURN_OFF", "turn_off")
+            self.__mod_supported_features |= (  # pylint: disable=unused-private-member
+                ClimateEntityFeature.TURN_OFF
+            )
+
+        if not self.supported_features & ClimateEntityFeature.TURN_ON and (
+            type(self).async_turn_on is not ClimateEntity.async_turn_on
+            or type(self).turn_on is not ClimateEntity.turn_on
+        ):
+            # turn_on implicitly supported by implementing turn_on method
+            _report_turn_on_off("TURN_ON", "turn_on")
+            self.__mod_supported_features |= (  # pylint: disable=unused-private-member
+                ClimateEntityFeature.TURN_ON
+            )
+
+        if (
+            self.supported_features & ClimateEntityFeature.TURN_OFF
+            or self.supported_features & ClimateEntityFeature.TURN_ON
```

Comment:
Let's remove this check and instead rely on integrations indicating that they've been migrated